### PR TITLE
more info in crash reports

### PR DESCRIPTION
**Pull request by DeltaNedas**
add oh no, mindustry version, mod list and a link to create a bug report (using template)

### CODE REVIEW

**Comment by sk7725** [comment]
Oh no, Mindustry crashed!\n" + "Please report this at https://github.com/Anuken/Mindustry/issues/new?labels=bug&template=bug_report.md\n\n" 

This...is not a good idea...at least the wording.
If a mod crashes some poorly-educated people/mod owner will head straight to Anuke, when the issue is the mod and not the game(or does Anuke accept mod issues?)

**Comment by sk7725** [comment]
Also try making it that it uses bundles and not hardcoded text?

**Comment by DeltaNedas** [comment]
anuke would then need to translate the crash log to make use of it
like with translating official / release /etc in the version

**Comment by DeltaNedas** [comment]
modded reports wont suggest to report crashes now

**Comment by DeltaNedas** [comment]
@Anuken

**Comment by Anuken** [comment]
If you're going to add more info to crash reports, there needs to be a lot more than this. See what I send to the crash report server, for example.

**Comment by DeltaNedas** [comment]
ok ill add more

**Comment by DeltaNedas** [comment]
@Anuken  any other info that should be added?

File: core/src/mindustry/net/CrashSender.java [hunk 1]
```diff
@@ -19,19 +19,33 @@
...
 
 public class CrashSender{
 
+    public static String createReport(String error){
+        String report = "Oh no, Mindustry crashed!\n";
+        if(mods.list().size == 0){
+            report += "Please report this at https://github.com/Anuken/Mindustry/issues/new?labels=bug&template=bug_report.md\n\n";
+        }
+        return report + "Version: " + Version.combined() + (Vars.headless ? " (Server)" : "") + "\n"
+            + "OS: " + System.getProperty("os.name") + "x" + (OS.is64Bit ? "64" : "32") + "\n"
```
Comment (Anuken) [inline core/src/mindustry/net/CrashSender.java:33]:
```suggestion
            + "OS: " + System.getProperty("os.name") + " x" + (OS.is64Bit ? "64" : "32") + "\n"
```

File: core/src/mindustry/net/CrashSender.java [hunk 2]
```diff
@@ -19,19 +19,33 @@
...
+            + "Java Version: " + System.getProperty("java.version") + "\n"
+            + "Java Architecture: " + System.getProperty("sun.arch.data.model") + "\n"
+            + mods.list().size + " Mods: " + mods.list().toString(", ", mod -> mod.name)
```
Comment (Anuken) [inline core/src/mindustry/net/CrashSender.java:36]: mod version would be nice too, e.g. `examplemod:1.0`

File: core/src/mindustry/net/CrashSender.java [hunk 3]
```diff
@@ -130,11 +144,11 @@ public static void send(Throwable exception, Cons<File> writeListener){
             boolean[] sent = {false};
 
             Log.info("Sending crash report.");
-            //post to crash report URL
+            //post to crash report URL, exit code indicates send success
             httpPost(Vars.crashReportURL, value.toJson(OutputType.json), r -> {
                 Log.info("Crash sent successfully.");
                 sent[0] = true;
-                System.exit(1);
+                System.exit(0);
```
Comment (Anuken) [inline core/src/mindustry/net/CrashSender.java:151]: But the application still *crashed*, it should exit with 1 or at least a non-zero code.

**Review by Anuken — CHANGES_REQUESTED** [review]

**Comment by DeltaNedas** [comment]
@Anuken done

**Review by DeltaNedas — COMMENTED** [review]
done pls go away bad box

**Pull request merged**